packages/map-template/src/helpers/GetDesktopPaddingBottom.js

```diff
@@ -1,9 +1,27 @@
 /**
  * Get bottom padding on desktop.
+ * Returns a promise with the padding in pixels.
+ *
+ * @returns {Promise}
  */
 export default function getDesktopPaddingBottom() {
-    const sidebar = document.querySelector('.modal--open');
-    const mapContainer = document.querySelector('.mapsindoors-map');
-    // Subtract the top padding from the height of the map container element.
-    return mapContainer.offsetHeight - sidebar?.offsetTop;
-}
\ No newline at end of file
+    return new Promise(resolve => {
+        const sidebar = document.querySelector('.modal--open');
+        const mapContainer = document.querySelector('.mapsindoors-map');
+
+        // If there is no sidebar element (yet), setup a MutationObserver that listens for the appearance of it.
+        if (!sidebar) {
+            const observer = new MutationObserver(() => {
+                const sidebar = document.querySelector('.modal--open');
+                if (sidebar) {
+                    observer.disconnect();
+                    resolve(mapContainer.offsetHeight - sidebar?.offsetTop); // Subtract the top padding from the height of the map container element.
+                }
+            });
+
+            observer.observe(document, { attributes: false, childList: true, characterData: false, subtree: true });
```

Comment:
Can you put a few more words to why this is needed? 🙏🏻

Reply:
I tried in line 12. Or is it this specific line you want some elaboration on?

Reply:
Yes, I was mostly asking about the props inside the object, childList, characterData, subtree

Reply:
I'll see what I can do. If anything, I can link to https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver/observe#options

Reply:
That's what I was looking for, perfect 😄

Reply:
Done.